x/auth/client/cli/tx_multisign.go

```diff
@@ -196,10 +189,208 @@ func makeMultiSignCmd() func(cmd *cobra.Command, args []string) (err error) {
+			txFactory = txFactory.WithSignMode(signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON)
+		}
+
+		var infile = os.Stdin
+		if args[0] != "-" {
+			infile, err = os.Open(args[0])
+			defer func() {
+				err2 := infile.Close()
+				if err == nil {
+					err = err2
+				}
+			}()
+
+			if err != nil {
+				return fmt.Errorf("couldn't open %s: %w", args[0], err)
+			}
+		}
+		scanner := authclient.NewBatchScanner(txCfg, infile)
+
+		multisigInfo, err := getMultisigInfo(clientCtx, args[1])
+		if err != nil {
+			return err
+		}
+
+		var signatureBatch [][]signingtypes.SignatureV2
+		for i := 2; i < len(args); i++ {
+			sigs, err := readSignaturesFromFile(clientCtx, args[i])
+			if err != nil {
+				return err
+			}
+
+			signatureBatch = append(signatureBatch, sigs)
+		}
+
+		if !clientCtx.Offline {
+			accnum, seq, err := clientCtx.AccountRetriever.GetAccountNumberSequence(clientCtx, multisigInfo.GetAddress())
+			if err != nil {
+				return err
+			}
+
+			txFactory = txFactory.WithAccountNumber(accnum).WithSequence(seq)
+		}
+
+		for i := 0; scanner.Scan(); i++ {
+			txBldr, err := txCfg.WrapTxBuilder(scanner.Tx())
+			if err != nil {
+				return err
+			}
+
+			multisigPub := multisigInfo.GetPubKey().(*kmultisig.LegacyAminoPubKey)
+			multisigSig := multisig.NewMultisig(len(multisigPub.PubKeys))
+			signingData := signing.SignerData{
+				ChainID:       txFactory.ChainID(),
+				AccountNumber: txFactory.AccountNumber(),
+				Sequence:      txFactory.Sequence(),
+			}
+
+			for _, sig := range signatureBatch {
+				err = signing.VerifySignature(sig[i].PubKey, signingData, sig[i].Data, txCfg.SignModeHandler(), txBldr.GetTx())
+				if err != nil {
+					return fmt.Errorf("couldn't verify signature: %w %v", err, sig)
+				}
+
+				if err := multisig.AddSignatureV2(multisigSig, sig[i], multisigPub.GetPubKeys()); err != nil {
+					return err
+				}
+			}
+
+			sigV2 := signingtypes.SignatureV2{
+				PubKey:   multisigPub,
+				Data:     multisigSig,
+				Sequence: txFactory.Sequence(),
+			}
+
+			err = txBldr.SetSignatures(sigV2)
+			if err != nil {
+				return err
+			}
+
+			sigOnly, _ := cmd.Flags().GetBool(flagSigOnly)
+			aminoJSON, _ := cmd.Flags().GetBool(flagAmino)
+
+			var json []byte
+
+			if aminoJSON {
+				stdTx, err := tx.ConvertTxToStdTx(clientCtx.LegacyAmino, txBldr.GetTx())
+				if err != nil {
+					return err
+				}
+
+				req := rest.BroadcastReq{
+					Tx:   stdTx,
+					Mode: "block|sync|async",
+				}
+
+				json, _ = clientCtx.LegacyAmino.MarshalJSON(req)
+
+			} else {
+				json, err = marshalSignatureJSON(txCfg, txBldr, sigOnly)
+				if err != nil {
+					return err
+				}
+			}
+
+			err = clientCtx.PrintString(fmt.Sprintf("%s\n", json))
+			if err != nil {
+				return err
+			}
+
+			if viper.GetBool(flagNoAutoIncrement) {
+				continue
+			}
+			sequence := txFactory.Sequence() + 1
+			txFactory = txFactory.WithSequence(sequence)
+		}
+
+		return nil
+	}
+}
```

Comment:
This function is almost 150 lines long. It's too long and should be broken down.